extensions/sql-migration/src/wizard/skuRecommendationPage.ts

```diff
@@ -307,10 +306,10 @@ export class SKURecommendationPage extends MigrationWizardPage {
 				]
 			});
 
-			let dialog = new SkuRecommendationResultsDialog(this.migrationStateModel, product.type, recommendationsJSON);
```

Comment:
@rchlkm, from my understanding it seemed the only change I needed to make in order to link the backend SKU recommendation results with your UI was to replace this placeholder `recommendationsJSON` object with an actual `SkuRecommendationResult` that came from a call to the backend, which is what I originally tried. 

However the problem I ran into is that this code here runs as soon as the page is loaded (in fact, as soon as the wizard is loaded) and so of course there is no recommendation results to use yet. So instead I modified it so that the `SkuRecommendationResult` is passed in at the time the dialog is opened (as part of `dialog.openDialog()`) instead of at the time the dialog is created, which seems to work fine.

I'm not sure if this is the best way to make it work, so could you take a look at my changes and see if it makes sense?

Reply:
this makes sense. thanks for refactoring the code